pkg/database/email_config.go

```diff
@@ -0,0 +1,123 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package database
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/google/exposure-notifications-verification-server/pkg/email"
+	"github.com/jinzhu/gorm"
+)
+
+// EmailConfig represents and email configuration.
+type EmailConfig struct {
+	gorm.Model
+	Errorable
+
+	// email Config belongs to exactly one realm.
+	RealmID uint `gorm:"type:integer"`
+
+	// ProviderType is the email provider type - it's used to determine the
+	// underlying configuration.
+	ProviderType email.ProviderType `gorm:"type:varchar(100)"`
+
+	SMTPAccount string `gorm:"type:varchar(250)"`
+	SMTPHost    string `gorm:"type:varchar(250)"`
+	SMTPPort    string `gorm:"type:varchar(250)"`
+
+	// TwilioAuthToken is encrypted/decrypted automatically by callbacks. The
+	// cache fields exist as optimizations.
+	SMTPPassword                string `gorm:"type:varchar(250)"`
```

Comment:
Yea, here. Make sure we add `json:"-"` to prevent zap from logging it

Reply:
do we want that on this field's twin "TwilioAuthToken"

Reply:
ya, i probably forgot to add it.